sdk/storage/inc/common/unused/StorageUriBuilder.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <string>
+
+#include "azure/core/Uri.h"
+
+namespace Azure { namespace Storage {
+
+class StorageUriBuilder
+{
+public:
+    explicit StorageUriBuilder(Azure::Core::Http::Uri uri);
+
+    const std::string& GetScheme() const;
+    void SetScheme(std::string scheme);
+
+    const std::string& GetHost() const;
+    void SetHost(std::string host);
+
+    const std::string& GetPath const;
+    void SetPath(std::string path);
+
+    const std::string& GetQuery() const;
+    void SetQuery(std::string query);
+
+    uint16_t GetPort() const;
+    void SetPort(uint16_t port);
+
+    const std::string& GetAccountName() const;
```

Comment:
This is very dangerous. The URL doesn't have to have an account name in it; think CNAMEs.